### PR TITLE
[TensorIR] CreatePrimFunc from TE

### PR DESCRIPTION
**Pull request by Hzfengsy**
This PR introduces a method to create PrimFunc (and further TensorIR schedule) from Tensor Expression. (e.g. `te.compute`)

cc @tqchen @junrushao1994 @comaniac @jcf94 @xqdan

### CODE REVIEW

**Comment by tkonolige** [comment]
Would it be possible to add a name field to the `create_prim_func` function? That way when we print out primfuncs for debugging, they aren't all called "main".

**Comment by tqchen** [comment]
Create prim_func returns a PrimFunc, the name is attached in module creation process, we can likely recompose by

```
mod = IRModule({"mycoolname": create_prim_func())
```

File: python/tvm/te/operation.py [hunk 1]
```diff
@@ -426,3 +427,51 @@ def reduce_axis(dom, name="rv", thread_tag="", span=None):
...
+        func = create_prim_func([A, B, C])
+        print(tvm.script.asscript(func))
+
+    If we want to use TensorIR schedule to do transformations on such kernel,
+    we need to use `create_prim_func([A, B, C])` to create a schedulable PrimFunc.
+    The generated function looks like:
+
+    .. code-block:: python
+
+        def tir_matmul(a: ty.handle, b: ty.handle, c: ty.handle) -> None:
```
Comment (tqchen) [inline python/tvm/te/operation.py:460]: add a @tvm.script.tir decorator

**Review by tqchen — CHANGES_REQUESTED** [review]

**Comment by tkonolige** [comment]
Instead of `create_prim_func().with_attr("tir.global_symbol", global_symbol)`, I think we should just make `create_prim_func` take a name parameter. I was working on autoscheduler earlier and all the primfuncs it had were called "main". Having a name parameter encourages the user to name their functions and makes debugging easier.

**Comment by tqchen** [comment]
I think we can add `global_symbol` as an optional argument(default to main)

**Comment by junrushao1994** [comment]
@tkonolige The mechanism causing the issue you mentioned is that when constructing a `tir::Schedule` class, and the input is a PrimFunc, then it will put that PrimFunc into an IRModule with the default name `main`, so that the schedule is always IRModule-based.

**Comment by junrushao1994** [comment]
@tqchen The problem of creating an IRModule instead is that we might want to an IRModule that contains multiple PrimFuncs, e.g.:

```python
mod = IRModule({
  "func_1": create_prim_func(...),
  "func_2": create_prim_func(...),
})
```

**Review by tqchen — APPROVED** [review]

**Comment by junrushao1994** [comment]
Thanks for the great discussion @tkonolige @tqchen @Hzfengsy! I am going to merge it in given there is no objection :-)

**Pull request merged**